paymentsheet/src/main/java/com/stripe/android/paymentsheet/repositories/StripeCustomerAdapter.kt

```diff
@@ -27,7 +28,21 @@ internal class StripeCustomerAdapter @Inject constructor(
     private var cacheDate: Long? = null
 
     override suspend fun retrievePaymentMethods(): Result<List<PaymentMethod>> {
-        TODO()
+        getCustomer().fold(
+            onSuccess = { customer ->
+                val paymentMethods = customerRepository.getPaymentMethods(
+                    customerConfig = PaymentSheet.CustomerConfiguration(
+                        id = customer.customerId,
+                        ephemeralKeySecret = customer.ephemeralKey
```

Comment:
Should we be validating the ephemeralKey in `getCustomer`?

Reply:
That is what I am wondering as well. I am not sure how to validate that the key is valid. @wooj-stripe, how does iOS validate that the key is valid?

Reply:
Is there any way to do client side key validation?  My guess is no, and that we should just try to make the network call with the key.  If it returns something then great.. if not, then present the error.